src/core/server/config/deprecation/core_deprecations.ts

```diff
@@ -136,6 +136,8 @@ export const coreDeprecationProvider: ConfigDeprecationProvider = ({ rename, unu
   unusedFromRoot('optimize.workers'),
   unusedFromRoot('optimize.profile'),
   unusedFromRoot('optimize.validateSyntaxOfNodeModules'),
+  unusedFromRoot('elasticsearch.preserveHost'),
+  unusedFromRoot('elasticsearch.startupTimeout'),
```

Comment:
Probably better to have it in the ES config deprecations instead: https://github.com/elastic/kibana/blob/bf04235dae35452061cc7ea3d86d96c19a58206c/src/core/server/elasticsearch/elasticsearch_config.ts#L126

Reply:
I was wondering about that, but my thinking was that deprecations are essential legacy or technical debt. They once belonged to a specific domain, but now they're just there to tell users they're doing something wrong until we make these config values invalid in 8.x. Instead of these distracting from the other domain code, having them isolated to one place is similar to how we isolated the `legacy` folder.

These other ES config deprecations require access to the ES config so they can't be centralised this way.